core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnReverse.java

```diff
@@ -0,0 +1,94 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND FREEDOM FROM
+ * INFRINGEMENT, AND ANY WARRANTY THAT THE DOCUMENTATION WILL CONFORM TO THE
+ * SOFTWARE, OR ANY WARRANTY THAT THE SOFTWARE WILL BE ERROR FREE.  IN NO EVENT
+ * SHALL NIST BE LIABLE FOR ANY DAMAGES, INCLUDING, BUT NOT LIMITED TO, DIRECT,
+ * INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES, ARISING OUT OF, RESULTING FROM,
+ * OR IN ANY WAY CONNECTED WITH THIS SOFTWARE, WHETHER OR NOT BASED UPON WARRANTY,
+ * CONTRACT, TORT, OR OTHERWISE, WHETHER OR NOT INJURY WAS SUSTAINED BY PERSONS OR
+ * PROPERTY OR OTHERWISE, AND WHETHER OR NOT LOSS WAS SUSTAINED FROM, OR AROSE OUT
+ * OF THE RESULTS OF, OR USE OF, THE SOFTWARE OR SERVICES PROVIDED HEREUNDER.
+ */
+
+package gov.nist.secauto.metaschema.core.metapath.function.library;
+
+import gov.nist.secauto.metaschema.core.metapath.DynamicContext;
+import gov.nist.secauto.metaschema.core.metapath.ISequence;
+import gov.nist.secauto.metaschema.core.metapath.MetapathConstants;
+import gov.nist.secauto.metaschema.core.metapath.function.FunctionUtils;
+import gov.nist.secauto.metaschema.core.metapath.function.IArgument;
+import gov.nist.secauto.metaschema.core.metapath.function.IFunction;
+import gov.nist.secauto.metaschema.core.metapath.item.IItem;
+import gov.nist.secauto.metaschema.core.util.ObjectUtils;
+
+import java.util.Collections;
+import java.util.List;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Implements <a href=
+ * "https://www.w3.org/TR/xpath-functions-31/#func-exists">fn:exists</a>.
+ */
+public final class FnReverse {
+  @NonNull
+  static final IFunction SIGNATURE_ONE_ARG = IFunction.builder()
+      .name("reverse")
+      .namespace(MetapathConstants.NS_METAPATH_FUNCTIONS)
+      .deterministic()
+      .contextIndependent()
+      .focusIndependent()
+      .argument(IArgument.builder()
+          .name("target")
+          .type(IItem.class)
+          .zeroOrMore()
+          .build())
+      .returnType(IItem.class)
+      .returnZeroOrMore()
+      .functionHandler(FnReverse::execute)
+      .build();
+
+  private FnReverse() {
+    // disable construction
+  }
+
+  @SuppressWarnings("unused")
+  @NonNull
+  private static ISequence<?> execute(@NonNull IFunction function,
+      @NonNull List<ISequence<?>> arguments,
+      @NonNull DynamicContext dynamicContext,
+      IItem focus) {
+    ISequence<?> target = FunctionUtils.asType(ObjectUtils.requireNonNull(arguments.get(0)));
+    return ISequence.of(fnReverse(target));
+  }
+
+  /**
+   * Identify if there is at least one item in the {@code sequence}.
+   *
+   * @param sequence
+   *          the sequence to check
+   * @return {@code true} if the sequence contains at least one item, or
+   *         {@code false} otherwise
+   */
+  public static List<? extends IItem> fnReverse(List<? extends IItem> target) {
+    if (target.size() <= 1) {
+      return target;
+    }
+    Collections.reverse(target);
+    return target;
```

Comment:
There is no guarantee that target is mutable. Because of this a copy of target should be made before calling `Collections.reverse`. This is also important, since this call modifies the target directly.

Reply:
Resolved with https://github.com/aj-stein-nist/metaschema-java/commit/63be2a97a8558900a52b58221936b98b632eccd0.